bigfunctions/array_z_scores.yaml

```diff
@@ -0,0 +1,30 @@
+type: function_sql
+category: transform_array
+author:
+  name: Julien Awonga
+  url: https://linkedin.com/in/julienawonga
+  avatar_url: "https://media.licdn.com/dms/image/D4E03AQFf9Qpnl0adhg/profile-displayphoto-shrink_400_400/0/1690023383639?e=1721260800&v=beta&t=StL_M-62Xv8rvUaHFxdFu4kCnKr-tSm7PDMkjKn72L8"
+description: |
+  Returns an array of z-scores, which are standardized values showing how many standard deviations each element is from the mean.
+arguments:
+  - name: arr
+    type: any type
+output:
+  name: array_z_scores
+  type: any type
+examples:
+  - description: ""
+    arguments:
+      - "[1, 2, 3, 4, 5]"
+    output: "[−1.414,−0.707,0,0.707,1.414]"
+code: |
+  (
+    WITH stats AS (
+    SELECT 
+      avg(value) AS mean, 
+      stddev_pop(value) AS stddev
+    FROM unnest(arr) AS value
+    )
+    SELECT array_agg((value - stats.mean) / stats.stddev)
+    FROM unnest(arr) AS value, stats
+  )
```

Comment:
I think the convention of this repo is to have lower case SQL clauses :)

```sql
  (
    with stats as (
      select 
        avg(value) as mean, 
        stddev_pop(value) as stddev
      from unnest(arr) as value
    )
    select array_agg((value - stats.mean) / stats.stddev)
    from unnest(arr) as value, stats
  )
```

Given this, being explicit is always better so you can change 
```sql
    from unnest(arr) as value, stats
```
by 
```sql
    from unnest(arr) as value
    cross join stats
```

And if your standard deviation is `0` you will have to a division by zero error. So maybe replace you `SELECT` expression by the following SQL to handle this exceptional case (`safe_divide` returns `null` if your denominator is `0`. And as `null` values are not authorized in arrays, the `coalesce` reset the default value to `0`)


```sql
    select array_agg(coalesce(safe_divide(value - stats.mean,  stats.stddev), 0))
```

Reply:
agree with everything you wrote @AxelThevenot.

I will merge the PR and make the changes you suggested before deploying.